packages/signers/src/sign-transaction.ts

```diff
@@ -1,42 +1,38 @@
 import { SOLANA_ERROR__SIGNER__TRANSACTION_SENDING_SIGNER_MISSING, SolanaError } from '@solana/errors';
 import { SignatureBytes } from '@solana/keys';
+import { CompilableTransactionMessage } from '@solana/transaction-messages';
 import {
-    assertTransactionIsFullySigned,
-    CompilableTransaction,
-    IFullySignedTransaction,
-    ITransactionWithSignatures,
+    compileTransaction,
+    FullySignedTransaction,
+    newAssertTransactionIsFullySigned,
+    NewTransaction,
 } from '@solana/transactions';
 
-import { getSignersFromTransaction, ITransactionWithSigners } from './account-signer-meta';
+import { getSignersFromTransaction, ITransactionMessageWithSigners } from './account-signer-meta';
 import { deduplicateSigners } from './deduplicate-signers';
 import { isTransactionModifyingSigner, TransactionModifyingSigner } from './transaction-modifying-signer';
 import { isTransactionPartialSigner, TransactionPartialSigner } from './transaction-partial-signer';
 import { isTransactionSendingSigner, TransactionSendingSigner } from './transaction-sending-signer';
 import { isTransactionSigner, TransactionSigner } from './transaction-signer';
-import { ITransactionWithSingleSendingSigner } from './transaction-with-single-sending-signer';
+import { ITransactionMessageWithSingleSendingSigner } from './transaction-with-single-sending-signer';
 
-type CompilableTransactionWithSigners = CompilableTransaction &
-    ITransactionWithSigners &
-    Partial<ITransactionWithSignatures>;
+type CompilableTransactionMessageWithSigners = CompilableTransactionMessage & ITransactionMessageWithSigners;
 
 /**
  * Signs a transaction using any signers that may be stored in IAccountSignerMeta instruction accounts
  * as well as any signers provided explicitly to this function.
  * It will ignore TransactionSendingSigners since this function does not send the transaction.
  */
-export async function partiallySignTransactionWithSigners<
-    TTransaction extends CompilableTransactionWithSigners = CompilableTransactionWithSigners,
->(
-    transaction: TTransaction,
-    config: { abortSignal?: AbortSignal } = {},
-): Promise<ITransactionWithSignatures & TTransaction> {
+export async function partiallySignTransactionMessageWithSigners<
+    TTransactionMessage extends CompilableTransactionMessageWithSigners = CompilableTransactionMessageWithSigners,
+>(transactionMessage: TTransactionMessage, config: { abortSignal?: AbortSignal } = {}): Promise<NewTransaction> {
     const { partialSigners, modifyingSigners } = categorizeTransactionSigners(
-        deduplicateSigners(getSignersFromTransaction(transaction).filter(isTransactionSigner)),
+        deduplicateSigners(getSignersFromTransaction(transactionMessage).filter(isTransactionSigner)),
```

Comment:
Isn't this suppose to use `getSignersFromTransactionMessage` now?

```suggestion
        deduplicateSigners(getSignersFromTransactionMessage(transactionMessage).filter(isTransactionSigner)),
```